File: 프로그래머스/기초 트레이닝 - Level 0/2의 영역-solve.py
def solution(arr):
    is_2_in = False
    start = 0
    end = len(arr) - 1
    for i in range(len(arr)):
        if arr[i] == 2:
            start = i
            is_2_in = True 
            break
            
    for i in range(len(arr) - 1, -1, -1):
        if arr[i] == 2:
            end = i
            break
            
    return arr[start : end + 1] if is_2_in else [-1]

File: 프로그래머스/기초 트레이닝 - Level 0/2의 영역-solve.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr) {
        boolean is2In = false;
        int start = 0;
        int end = arr.length - 1;
        
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == 2) {
                start = i;
                is2In = true;
                break; 
            }
        }
        
        for (int i = arr.length - 1; i >= 0; i--) {
            if (arr[i] == 2) {
                end = i;
                break; 
            }
        }
        
        return is2In ? Arrays.copyOfRange(arr, start, end + 1) : new int[]{-1};
    }
}